src/subdomains/supporting/payment/services/transaction-helper.ts

```diff
@@ -434,6 +441,80 @@ export class TransactionHelper implements OnModuleInit {
     };
   }
 
+  async getInvoiceDetails(userDataId: number, txId: number): Promise<TxStatementDetails> {
+    return this.getTxStatementDetails(userDataId, txId, TxStatementType.INVOICE);
+  }
+
+  async getReceiptDetails(userDataId: number, txId: number): Promise<TxStatementDetails> {
+    return this.getTxStatementDetails(userDataId, txId, TxStatementType.RECEIPT);
+  }
+
+  private async getTxStatementDetails(
+    userDataId: number,
+    txId: number,
+    statementType: TxStatementType,
+  ): Promise<TxStatementDetails> {
+    const transaction = await this.transactionService.getTransactionById(txId, {
+      userData: true,
+      buyCrypto: { buy: true, cryptoRoute: true, cryptoInput: true },
+      buyFiat: { sell: true, cryptoInput: true },
+      refReward: { user: { userData: true } },
+    });
+
+    if (!transaction || !transaction.targetEntity || transaction.targetEntity instanceof BankTxReturn)
+      throw new BadRequestException('Transaction not found');
+    if (!transaction.userData.isDataComplete) throw new BadRequestException('User data is not complete');
+    if (!transaction.targetEntity.isComplete) throw new BadRequestException('Transaction not completed');
+    if (transaction.userData.id !== userDataId) throw new ForbiddenException('Not your transaction');
+
+    if (transaction.buyCrypto && !transaction.buyCrypto.isCryptoCryptoTransaction) {
+      const currency = (await this.fiatService.getFiatByName(transaction.buyCrypto.inputAsset)).name;
```

Comment:
I think I would either do 
`const currency = await this.fiatService.getFiatByName(transaction.buyCrypto.inputAsset)` and use the lines below `currency.name` 
or I would use 
`await this.fiatService.getFiatByName(transaction.buyCrypto.inputAsset).then((c) => c.name)`